source/source.go

```diff
@@ -80,14 +78,51 @@ func (s *Source) Configure(ctx context.Context, cfgRaw commonsConfig.Config) err
 func (s *Source) Open(ctx context.Context, position opencdc.Position) error {
 	sdk.Logger(ctx).Info().Msg("Opening an Amazon Redshift Source...")
 
-	pos, err := iterator.ParseSDKPosition(position)
+	var err error
+	s.position, err = ParseSDKPosition(position)
+	if err != nil {
+		return err
+	}
+
+	// Open database connection
+	s.db, err = sqlx.Open(driverName, s.config.DSN)
 	if err != nil {
-		return fmt.Errorf("parse position: %w", err)
+		return fmt.Errorf("failed opening db connection: %w", err)
 	}
 
-	s.iterator, err = iterator.New(ctx, driverName, pos, s.config)
+	// Check the connection
+	pingCtx, cancel := context.WithTimeout(ctx, pingTimeout)
+	defer cancel()
+	err = s.db.PingContext(pingCtx)
 	if err != nil {
-		return fmt.Errorf("new iterator: %w", err)
+		return fmt.Errorf("ping db with timeout: %w", err)
+	}
+
+	s.ch = make(chan opencdc.Record, s.config.BatchSize)
+	s.wg = &sync.WaitGroup{}
+
+	for table, tableConfig := range s.config.Tables {
+		s.wg.Add(1)
+
+		tablePosition, _ := s.position.get(table)
+		// A new iterator for each table
+		err := NewIterator(ctx, IteratorConfig{
+			db:             s.db,
+			position:       tablePosition,
+			sourcePosition: s.position,
+			table:          table,
+			orderingColumn: tableConfig.OrderingColumn,
+			keyColumns:     tableConfig.KeyColumns,
+			batchSize:      s.config.BatchSize,
+			pollingPeriod:  s.config.PollingPeriod,
+			snapshot:       s.config.Snapshot,
+			dsn:            s.config.DSN,
+			wg:             s.wg,
+			ch:             s.ch,
+		})
+		if err != nil {
+			return fmt.Errorf("new iterator: %w", err)
```

Comment:
It would be good to also mention what's the table for which we couldn't create an iterator.